mm/mmap.c

```diff
@@ -507,12 +509,7 @@ int do_mmap(struct inode *i, unsigned int start, unsigned int length, unsigned i
 
 	if(i && i->fsop->mmap) {
 		if((errno = i->fsop->mmap(i, vma))) {
-			int errno2;
-
-			if((errno2 = free_vma_region(vma, start, length))) {
-				kfree((unsigned int)vma);
-				return errno2;
-			}
+			free_vma_region(vma, start, length);
```

Comment:
Beware, the function `free_vma_region()` can return `-ENOMEM`, but with this change we'll never know.

Reply:
I had considered that, but then realized 1) `free_vma_region` will never return `-ENOMEM` in this case, as it is always passed the entire vma region just created (so no split and no memory alloc), and 2) I thought it better to return the driver `mmap` subfunction error code to the application, rather than a possible `-ENOMEM` obscuring the real error.

Reply:
I see, OK.